apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/PublicAccessPreventionTest.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Storage.v1.Data;
+using Xunit;
+
+namespace Google.Cloud.Storage.V1.IntegrationTests
+{
+    [Collection(nameof(StorageFixture))]
+    public class PublicAccessPreventionTest
+    {
+        private const string EnforcedValue = "enforced";
+        private const string UnspecifiedValue = "unspecified";
+        private static readonly Policy.BindingsData AllUsersViewer = new Policy.BindingsData
+        {
+            Members = new[] { "allUsers" },
+            Role = "roles/storage.objectViewer"
+        };
+
+        private readonly StorageFixture _fixture;
+
+        public PublicAccessPreventionTest(StorageFixture fixture) => _fixture = fixture;
+
+        [Fact]
+        public void PreventAccessOnExistingBucket()
+        {
+            var client = _fixture.Client;
+            string bucketName = _fixture.GenerateBucketName();
+            Bucket bucket = _fixture.CreateBucket(bucketName, false);
+            Assert.Null(bucket.IamConfiguration.PublicAccessPrevention);
+
+            // Enforce PAP
+            client.PatchBucket(CreateBucketRepresentation(bucketName, EnforcedValue));
+
+            // We shouldn't be able to all users to view objects.
```

Comment:
nit: Incomplete comment.

Reply:
Whoops, thanks. Will just add "allow" after the "to"